### PR TITLE
added application/merge-patch+json to allowed json content types

### PR DESCRIPTION
**Pull request by jchannon**
This addresses #1937

### CODE REVIEW

File: src/Nancy/Json/Json.cs [hunk 1]
```diff
@@ -118,10 +118,11 @@ public static bool IsJsonContentType(string contentType)
...
             return contentMimeType.Equals("application/json", StringComparison.OrdinalIgnoreCase) ||
-                   contentMimeType.StartsWith("application/json-", StringComparison.OrdinalIgnoreCase) ||
-                   contentMimeType.Equals("text/json", StringComparison.OrdinalIgnoreCase) ||
-                  (contentMimeType.StartsWith("application/vnd", StringComparison.OrdinalIgnoreCase) &&
-                   contentMimeType.EndsWith("+json", StringComparison.OrdinalIgnoreCase));
+            contentMimeType.StartsWith("application/json-", StringComparison.OrdinalIgnoreCase) ||
+            contentMimeType.Equals("text/json", StringComparison.OrdinalIgnoreCase) ||
+            (contentMimeType.StartsWith("application/vnd", StringComparison.OrdinalIgnoreCase) &&
+            contentMimeType.EndsWith("+json", StringComparison.OrdinalIgnoreCase)) ||
+            contentMimeType.Equals("application/merge-patch+json");
```
Comment (khellang) [inline src/Nancy/Json/Json.cs]: Isn't this a bit specific? And redundant? The `EndsWith("+json")` should cover both cases.

**Comment by khellang** [comment]
Ping @jchannon!

File: src/Nancy/Json/Json.cs (hunk 1 above)
Comment (jchannon) [inline src/Nancy/Json/Json.cs]: so remove

```
(contentMimeType.StartsWith("application/vnd", StringComparison.OrdinalIgnoreCase) &&
contentMimeType.EndsWith("+json", StringComparison.OrdinalIgnoreCase)) ||
contentMimeType.Equals("application/merge-patch+json")
```

and just make it `contentMimeType.EndsWith("+json", StringComparison.OrdinalIgnoreCase)`

Comment (grumpydev) [inline src/Nancy/Json/Json.cs]: I thought it already handled +json actually

Comment (khellang) [inline src/Nancy/Json/Json.cs]: @grumpydev It's `StartsWith("application/vnd") && EndsWith("+json")`.

@jchannon Yeah. Let's just check `EndsWith("+json")`. I think that should be enough. Remember to update the XML doc :wink:

**Comment by jchannon** [comment]
Re-check

**Pull request merged**